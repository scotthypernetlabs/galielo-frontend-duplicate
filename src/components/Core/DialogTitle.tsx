import { Close } from "@material-ui/icons";
import { IconButton, Typography } from "@material-ui/core";
import MuiDialogTitle from "@material-ui/core/DialogTitle/DialogTitle";
import React from "react";

interface DialogTitleProps {
  title: string;
  handleClose: any;
}

const DialogTitle: React.SFC<DialogTitleProps> = (props: DialogTitleProps) => {
  const { title, handleClose } = props;
  return (
    <MuiDialogTitle disableTypography>
      <Typography variant="h3" className="typography">
        {title}
      </Typography>
      <IconButton
        onClick={handleClose}
        style={{ position: "absolute" }}
        className="close-button add-cursor"
      >
        <Close />
      </IconButton>
    </MuiDialogTitle>
  );
};

export default DialogTitle;
