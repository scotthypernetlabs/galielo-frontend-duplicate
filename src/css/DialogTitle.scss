.typography {
  font-weight: 700;
}

.close-button {
  right: 5px;
  top: 5px;
}
