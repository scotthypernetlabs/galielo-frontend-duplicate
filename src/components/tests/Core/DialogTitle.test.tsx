import { IconButton, Typography } from "@material-ui/core";
import { mount } from "enzyme";
import DialogTitle from "../../Core/DialogTitle/DialogTitle";
import React from "react";

describe("DialogTitle Component", () => {
  const componentProps = {
    title: "Testing Title",
    handleClose: jest.fn()
  };

  const Wrapper = (props: any) => {
    return <DialogTitle {...componentProps} />;
  };

  const dialogTitle = mount(<Wrapper />);

  it("should display title", () => {
    expect(dialogTitle.find(Typography)).toHaveLength(1);
    expect(
      dialogTitle
        .find(Typography)
        .childAt(0)
        .text()
    ).toEqual(componentProps.title);
  });

  it("icon should call on click", () => {
    expect(dialogTitle.find(IconButton)).toHaveLength(1);
    dialogTitle.find(IconButton).simulate("click");
    expect(dialogTitle.childAt(0).props().handleClose).toHaveBeenCalled();
  });

  it("should match snapshot", () => {
    expect(dialogTitle).toMatchSnapshot();
  });
});
