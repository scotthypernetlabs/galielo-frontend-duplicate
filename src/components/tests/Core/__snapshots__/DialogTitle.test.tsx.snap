// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogTitle Component should match snapshot 1`] = `ReactWrapper {}`;
